.changeset/dull-tools-knock.md

```diff
@@ -0,0 +1,5 @@
+---
+'@guardian/libs': major
```

Comment:
Is this a breaking change or just a feature?

Reply:
I think it's a breaking change because of exhaustiveness checking? Any code that previously checked all variations of `Theme` will now have an extra variation that it's not checking for. For example:

```ts
const themeToString = (theme: ArticleTheme): string => {
  switch (theme) {
    case ArticlePillar.News:
      return 'news';
    case ArticlePillar.Sport:
      return 'sport';
    case ArticleSpecial.Labs:
      return 'labs';
    ...
  }
}
```

There's no case for `ArticleSpecial.SpecialReportAlt`, so the compiler will generate an error. It has to because otherwise the new case wouldn't be matched, resulting in `undefined` being implicitly returned, which is not equal to `string`.

Reply:
Yes, exactly what @JamieB-gu says. We do have places in DCR where we switch over `format.theme` without having a default case, e.g. https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/lib/decidePalette.ts#L202-L217

At the moment TS is not complaining but we have a PR to fix this: https://github.com/guardian/dotcom-rendering/pull/6105.

Reply:
Thanks for clarifying 👍